cli/mender-artifact/write.go

```diff
@@ -517,10 +528,28 @@ func getDeviceSnapshot(c *cli.Context) (string, error) {
 		args = append(args, "-p", port)
 	}
 	args = append(args, userAtHost)
+	useSudo, _ := hasCommand(append(args, "sudo", "--help"))
+
+	_, out := hasCommand(append(args, "id", "-u"))
+	if string(out) == "0" {
+		useSudo = false
+		Log.Debug("effective user id on host is 0; not using sudo.")
+	}
+
+	if strings.HasPrefix(userAtHost, "root@") {
```

Comment:
This should be an `else if` so as not to make the log entries confusing.

Reply:
sure. thanks.